cmake/DependencyManager.cmake

```diff
@@ -666,6 +666,33 @@ function(__DependencyManager_VersionCheck versionRange version out)
     set(${out} "${compatible}" PARENT_SCOPE)
 endfunction()
 
+#[=======================================================================[.rst:
+.. code-block:: cmake
+
+    __DependencyManager_getProjectName(<path> <out>)
+
+Determines the name of the first encountered project() call in the file pointed to by
+the provided ``<path>`` and writes the determined name into ``<out>`` (in the parent scope).
+If the parsing has failed or the given file does not contain a project() invocation, then
+``<out>`` is set to an empty string.
+#]=======================================================================]
+function(__DependencyManager_getProjectName path outVar)
+    messagev("__DependencyManager_getProjectName from '${path}' to ${outVar}")
+
+    file(READ "${path}" contents)
+
+    # Strip comments (ignoring potential issues due to hashes in strings)
+    string(REGEX REPLACE "#[^\r\n]*" "" contents "${contents}")
+
+    if ("${contents}" MATCHES "project[ \t\r\n]*\\([ \t\r\n]*([a-zA-Z_0-9-]+)")
```

Comment:
cmake commands are case insensitive, so it might not be `project`, but instead `Project` or `PROJECT` or `pROJECT`.  A simple fix might be to put the three alternates (not `pROJECT`!) in the regex.

Reply:
I have adapted the regex to match "project" case-insensitively.